File: DotNet/Report/Entities/ReportScheduleModel.cs

using LantanaGroup.Link.Shared.Application.Enums;
using LantanaGroup.Link.Shared.Application.Models;
using LantanaGroup.Link.Shared.Domain.Attributes;
using LantanaGroup.Link.Shared.Domain.Entities;
using MongoDB.Bson.Serialization.Attributes;

namespace LantanaGroup.Link.Report.Entities
{

    [BsonCollection("reportSchedule")]
    [BsonIgnoreExtraElements]
    public class ReportScheduleModel : BaseEntityExtended
    {
        public string FacilityId { get; set; } = string.Empty;
        public DateTime ReportStartDate { get; set; }
        public DateTime ReportEndDate { get; set; }
        public DateTime? SubmitReportDateTime { get; set; } = null;
        public bool EnableSubmission { get; set; } = true;
        public bool EndOfReportPeriodJobHasRun { get; set; } = false;
        public List<string> ReportTypes { get; set; } = new List<string>();
        public Frequency Frequency { get; set; }
        public string? PayloadRootUri { get; set; }
        public ScheduleStatus Status { get; set; } = ScheduleStatus.New;
    }
}

File: Java/shared/src/main/java/com/lantanagroup/link/shared/entities/ReportScheduleModel.java
package com.lantanagroup.link.shared.entities;

import lombok.Getter;
import lombok.Setter;

import java.util.ArrayList;
import java.util.Date;
import java.util.List;

@Getter
@Setter
public class ReportScheduleModel {
    private String id;
    private Date createDate;
    private Date modifyDate;
    private String facilityId;
    private Date reportStartDate;
    private Date reportEndDate;
    private Date submitReportDateTime;
    private boolean enableSubmission = true;
    private boolean endOfReportPeriodJobHasRun = false;
    private List<String> reportTypes = new ArrayList<>();
    private AdHocType adHocType;
    private Frequency frequency;
    private String payloadRootUri;
    private ScheduleStatus status = ScheduleStatus.New;

}
